completer.go

```diff
@@ -1,23 +1,43 @@
 package completer
 
-type Completer map[string]string
+import "fmt"
+
+type Completer struct {
+	aliases   map[string]string
+	originals map[string]struct{}
+}
 
 func NewCompleter() Completer {
-	return Completer(make(map[string]string))
+	return Completer{aliases: make(map[string]string), originals: make(map[string]struct{})}
 }
 
-func (c Completer) Add(s string) {
+func (c Completer) Add(s string) error {
+	a := c.aliases
```

Comment:
Rather than creating a copy of `c.aliases` here, just use `c.aliases`. The compiler will figure it out.

Reply:
Since in my previous proposal adding a substring of an already existing string was a considered an error, then to avoid inconsistent states I had to only commit the changes if the addition was successful. I'll change the behavior and remove the copy.